Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-93.ca-central-1.compute.internal
Sha: f1270055d6ffc82fb908699f31bf987e352403ec
Number of Error(s): 99
checking for libcrypto (by pkg-config).../home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14174/HPCC-Platform/system/httplib/httplib.h:335:41: error: template argument 1 is invalid
/home/centos/smoketest/PR-14174/HPCC-Platform/system/httplib/httplib.h:335:41: error: template argument 2 is invalid
/home/centos/smoketest/PR-14174/HPCC-Platform/system/httplib/httplib.h:336:28: error: \xe2\x80\x98Range\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14174/HPCC-Platform/system/httplib/httplib.h:336:33: error: template argument 1 is invalid
/home/centos/smoketest/PR-14174/HPCC-Platform/system/httplib/httplib.h:336:33: error: template argument 2 is invalid
/home/centos/smoketest/PR-14174/HPCC-Platform/system/httplib/httplib.h:352:3: error: \xe2\x80\x98Ranges\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ASRange\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/httplib/httplib.h:1510:22: error: \xe2\x80\x98class httplib::Stream\xe2\x80\x99 has no member named \xe2\x80\x98read\xe2\x80\x99; did you mean \xe2\x80\x98Stream\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/httplib/httplib.h:1647:14: error: \xe2\x80\x98handle_EINTR\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not name a type; did you mean \xe2\x80\x98ossl_ssize_t\xe2\x80\x99?
/home/centos/smoketest/PR-14174/HPCC-Platform/system/jlib/jsecrets.cpp:44:17: error: \xe2\x80\x98httplib_ssize_t\xe2\x80\x99 does not
 ... ( comment is too long, 122 lines, 22302 bytes, truncated) ... 
[Build error:](https://gist.github.com/638d19ebfe01a26cf2c5440538f822b9)
Build log [Build log](https://gist.github.com/HPCCSmoketest/638d19ebfe01a26cf2c5440538f822b9#file-result-20-09-16-12-11-10-log)

```

--- a/system/include/platform.h
+++ b/system/include/platform.h
@@ -179,7 +179,11 @@ typedef memsize_t rowsize_t;
 
 typedef unsigned __int64 off64_t;
 typedef int socklen_t;
+#ifdef _WIN64
+typedef __int64 ssize_t; // correct return to type for unix read/write/pread etc.
+#else
 typedef int ssize_t; // correct return to type for unix read/write/pread etc.
+#endif
 #define fpos_ht fpos_t
 
 typedef long double LDouble;

